libnavigation-base/src/main/java/com/mapbox/navigation/base/options/LocationPollingOptions.kt

```diff
@@ -0,0 +1,108 @@
+package com.mapbox.navigation.base.options
+
+import com.mapbox.android.core.location.LocationEngineRequest
+
+/**
+ * Defines options for the navigator polling loops.
+ *
+ * TODO move NavigationOptions.navigatorPredictionMillis to these options
+ * TODO move NavigationOptions.locationEngine to these options
+ *
+ * @param locationEngineRequest specifies the rate to request locations from the device.
+ * @param navigatorIntervalMillis the rate to update the navigator when no locations have been returned. Default value 1000L.
+ * @param navigatorPatienceMillis the time to wait for no locations, and then start a polling interval. Default value 2000L.
+ */
+class LocationPollingOptions private constructor(
```

Comment:
We were considering names here. Cause it could also be `TripSessionOptions`. Which could open up for more possibilities. But that is also making it something that it is not. I wouldn't expect `TripSessionOptions` to contain polling rates. Also the [TripSession interface is internal](https://github.com/mapbox/mapbox-navigation-android/blob/0ce0eb935242d0f90344cef9c95618f4af63305f/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSession.kt#L9)